src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -3085,14 +3085,25 @@ private void readExtra(ZipFileSystem zipfs) throws IOException {
                 int sz = SH(extra, pos + 2);
                 pos += 4;
                 if (pos + sz > elen) {        // invalid data
-                    throw new ZipException("Invalid CEN header (invalid zip64 extra data field size)");
+                    // There are some extra headers with an invalid data size, if
+                    // found, we need to just ignore them
+                    break;
                 }
                 switch (tag) {
                 case EXTID_ZIP64 :
                     // Check to see if we have a valid block size
                     if (!isZip64ExtBlockSizeValid(sz)) {
                         throw new ZipException("Invalid CEN header (invalid zip64 extra data field size)");
                     }
+                    // if ZIP64_EXTID blocksize == 0 validate csize, size and
+                    // locoff to make sure they do not == ZIP64_MINVAL
+                    if (sz == 0) {
+                        if ( csize == ZIP64_MINVAL || size == ZIP64_MINVAL ||
+                                locoff == ZIP64_MINVAL) {
+                            throw new ZipException("Invalid CEN header (invalid zip64 extra data field size)");
+                        }
+                        break;
+                    }
                     if (size == ZIP64_MINVAL) {
```

Comment:
Note that we always increase "pos" only in case of "_MINVAL". If the value of size and csize is correct in the "body" of the ziip file and only locoff is negative then we should skip two fields in the extra block and read the third one.